src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -460,6 +460,57 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  def getBilling(credential: Credential) = {
+    new Cloudbilling.Builder(httpTransport, jsonFactory, credential).build()
+  }
+
+  /**
+    * NOTE: This function will returns "false" in both of the following cases:
+    * if you don't have sufficient scopes
+    *   - Google's JSON response body will contain "message": "Request had insufficient authentication scopes."
+    * if you're not authorized to see the billing account
+    *   - Google's JSON response body will contain "message" : "The caller does not have permission"
+    */
+  protected def credentialOwnsBillingAccount(credential: Credential, billingAccountName: String): Future[Boolean] = {
+    val fetcher = getBilling(credential).billingAccounts().get(billingAccountName)
+    retryWithRecoverWhen500orGoogleError(() => {
+      blocking {
+        executeGoogleRequest(fetcher)
+      }
+      true //if the request succeeds, it has access.
+    }) {
+      case e: HttpResponseException if e.getStatusCode == StatusCodes.Forbidden.intValue => false
+    }
+  }
+
+  protected def listBillingAccounts(credential: Credential): Future[Seq[BillingAccount]] = {
+    val fetcher = getBilling(credential).billingAccounts().list()
+    retryWithRecoverWhen500orGoogleError(() => {
+      val list = blocking {
+        executeGoogleRequest(fetcher)
+      }
+      list.getBillingAccounts.toSeq
+    }) {
+      case e: HttpResponseException if e.getStatusCode == StatusCodes.Forbidden.intValue =>
+        throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, e.getContent))
```

Comment:
where does this tell the caller which scopes it needs? I don't think a 403 is enough - 403 could be thrown by the proxy for other reasons. I think that since rawls knows which scopes are necessary for this API call, rawls should indicate that in the error response. (see also Doug's comment on the jira ticket)